core.md

```diff
@@ -89,10 +162,35 @@ struct would:
     | y: float64 |      |
     +============+    --+
 
-Structure definitions can also reference other structure
-definitions:
+#### Indexed structs
+
+```js
+const PointPairType = new StructType(PointType, 2);
+```
+
+This defines a new type definition `LineType` that consists of two indexed
+elements, each an instance of `PointType`. These will be laid out in memory consecutively, just as a C struct would:
+
+    +===============+    --+ PointPairType
+    | 0: x: float64 |      | --+ PointType
+    |    y: float64 |      | --+
+    | 1: x: float64 |      |
+    |    y: float64 |      |
+    +===============+    --+
+
+An equivalent definition to this, that'd become unwieldy for large `length`s, would be:
+
+```js
+const PointPairType = new StructType({0: PointType, 1: PointType, length: 2});
```

Comment:
I changed this to say that `length` is defined as a property on the type's `prototype`. I actually think this is nicer, and it doesn't require either giving up on this equivalence or introducing singleton types.